hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResourceTest.kt

```diff
@@ -0,0 +1,660 @@
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.DateMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.IntMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.StringMatcher
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+class AttributeSearchResourceTest : AbstractSearchDataIntegrationTest() {
+  private fun validRequest(
+    matchers: String = """
+      {
+        "type": "String",
+        "attribute": "lastName",
+        "condition": "IS",
+        "searchTerm": "Smith"
+      }
+    """.trimIndent(),
+  ) = """{
+    "matchers": [
+      {
+        "joinType": "AND",
+        "matchers": [$matchers]
+      }
+    ]  
+  }
+  """.trimIndent()
+
+  private val today = LocalDate.now()
+  private val now = LocalDateTime.now()
+
+  @Nested
+  inner class Security {
+    @Test
+    fun `should be unauthorized without valid token`() {
+      webTestClient.post()
+        .uri("/attribute-search")
+        .header("Content-Type", "application/json")
+        .bodyValue(validRequest())
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `should be forbidden without a role`() {
+      webTestClient.post()
+        .uri("/attribute-search")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation())
+        .bodyValue(validRequest())
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should be forbidden with wrong role`() {
+      webTestClient.post()
+        .uri("/attribute-search")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation(roles = listOf("ROLE_UNKNOWN")))
+        .bodyValue(validRequest())
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should be OK with PRISONER_SEARCH role`() {
+      webTestClient.post()
+        .uri("/attribute-search")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_SEARCH")))
+        .bodyValue(validRequest())
+        .exchange()
+        .expectStatus().isOk
+    }
+
+    @Test
+    fun `should be OK with GLOBAL_SEARCH role`() {
+      webTestClient.post()
+        .uri("/attribute-search")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+        .bodyValue(validRequest())
+        .exchange()
+        .expectStatus().isOk
+    }
+  }
+
+  /**
+   * This class is to document the API behaviour and provide regression tests once that behaviour is settled.
+   *
+   * We need to document this behaviour because some of it isn't obvious - e.g. it's OK to pass an Int where a Boolean is expected. Also changing the JSON parser or configuring Jackson differently could change this behaviour. Which might be fine but it's good to know!
+   */
```

Comment:
It could be argued that these tests are just documenting the current Jackson behaviour. Hopefully this comment explains why I think they have value.